package day059;

public class ExceptionObject {
    public static void main(String[] args) {

        /*try catch block
		try{
			// code that might throw an exception
		}catch(Exception e){
			// just like method parameter :
			// Exception e  is a variable with type Exception
			// when java throw exception at run time
			// it will assign the exception object to variable e
			// for example if StringIndexOutOfBoundException happen in try block
			// it will do below
			// Exception e = new StringIndexOutOfBoundException("some message here");
			// if it's NullPointerException it will do below
			// Exception e = new NullPointerException(...);
			// since it's an object , we can call a method just like any other object
			// getMessage is useful method from Exception class

			try {
			    System.out.println(name.charAt(targetIndex));
                }catch(Exception e){
                System.out.println("Message by ....e.getMessage.... "+e.getMessage());
                 //since we have Exception object and its container "e"
                 // we can use pointer e to call getMessage();



					// to get some more information about the exception object
					// created at run time
			// if the exception happen the flow of program will come here
			// take action to recover or just report or notify
		}
		// some more code
		// by handling the exception might occure during the program
		// our program will not shut down abruptly
		// we can end or continue the program gracefully.
		or declare it .    public void doSomething() throws Exception{}
*/
    }
}
